src/raft.rs

```diff
@@ -1617,20 +1805,24 @@ impl<T: Storage> Raft<T> {
         if maybe_commit {
             if self.maybe_commit() {
                 if self.should_bcast_commit() {
-                    self.bcast_append();
+                    self.bcast_append(None);
                 }
             } else if old_paused {
                 // update() reset the wait state on this node. If we had delayed sending
                 // an update before, send it now.
                 send_append = true;
             }
         }
-
         if send_append {
             let from = m.from;
-            let mut prs = self.take_prs();
-            self.send_append(from, prs.get_mut(from).unwrap());
-            self.set_prs(prs);
+            if m.from_delegate == INVALID_ID {
+                let mut prs = self.take_prs();
+                self.send_append(from, &mut prs);
+                self.set_prs(prs);
+            } else {
+                let members = self.groups.get_members(m.from_delegate).cloned();
+                self.bcast_append(members.as_ref());
```

Comment:
`bcast_append` calls `send_append`, which checks members' delegate again. I believe it can be improved, maybe we can take a try?